packages/sdk-components-react/src/html-embed.tsx

```diff
@@ -83,16 +93,33 @@ const Placeholder = (props: ChildProps) => {
   );
 };
 
+const ClientOnly = (props: { children: ReactNode }) => {
+  const isServer = useSyncExternalStore(
```

Comment:
is it worth extracting it into a hook and commenting what this does with a link to that great article?